pkg/scale/decode.go

```diff
@@ -60,26 +60,23 @@ func indirect(dstv reflect.Value) (elem reflect.Value) {
 func Unmarshal(data []byte, dst interface{}) (err error) {
 	dstv := reflect.ValueOf(dst)
 	if dstv.Kind() != reflect.Ptr || dstv.IsNil() {
-		err = fmt.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
+		err = fmt.Errorf("%w: %T", ErrUnsupportedDestination, dst)
 		return
 	}
 
 	elem := indirect(dstv)
-	if err != nil {
-		return
-	}
 
 	buf := &bytes.Buffer{}
 	ds := decodeState{}
 	_, err = buf.Write(data)
 	if err != nil {
-		return
+		return err
 	}
 	ds.Reader = buf
```

Comment:
nit a bit out of scope, but this should really be `ds.Reader = bytes.NewBuffer(data)` so that:

1. no error to return or wrap or think about
2. no deep copy of the byte slice

I'd say go ahead do it here, or do it another PR, up to you :wink: